PR_Epic4_121_Созданы модели данных для фильтрации

sobolevkir opened the pull request:
Создан класс для всех параметров фильтрации - `FilterParameters`.

Upd: Скорректирован класс FilterParameters. Сделаны отдельные классы для параметров региона(Area может быть или страной, если parentId=null, или регионом, если parentId не равен null) и отрасли.

sobolevkir committed:
Созданы две модели данных - общий класс для всех параметров фильтров FilterParameters и класс одного параметра FilterParameter.

danil377377 reviewed (approved):
Ок, но тогда ещё нужно сделать FilterParameterDto, я могу сам его добавить.

danil377377 reviewed (changes requested):
А почему id не инт? У меня ответ с сервера через инт работает, так же будет менее затратно

sobolevkir commented:
> А почему id не инт? У меня ответ с сервера через инт работает, так же будет менее затратно

с Int будет проблема, т.к. ответы от сервера согласно API приходят в string и имеют точку в записи:
![image](https://github.com/user-attachments/assets/09cff4c1-abbd-45c6-b196-df5d77ff8eca)

sobolevkir committed:
Скорректирован класс FilterParameters.
Сделаны отдельные классы для параметров страны, региона и отрасли, поскольку в регионе должен быть ещё параметр parentId - id региона, в котором находится данный субъект.

sobolevkir committed:
Detekt formatting

sobolevkir commented:
> А почему id не инт? У меня ответ с сервера через инт работает, так же будет менее затратно

Да, он работает, т.к. эти параметры конвертируются автоматически. но по API там есть ещё точки в id например для отраслей. Поэтому лучше string

sobolevkir commented:
> Ок, но тогда ещё нужно сделать FilterParameterDto, я могу сам его добавить.

Теперь в этом нет необходимости. Сделал отдельные классы для Industry,Country и Region, как договорились, поскольку для региона нужен parentId. Можно конечно было бы сделать Nullable-поле parentId, но пока оставим отдельные классы лучше, чтобы не путаться)

sobolevkir committed:
Изменён класс FilterParameters:
для страны и региона сделан общий класс с Nullable-полем parentId.
